### PR TITLE
Improve encoder interface

### PR DESCRIPTION
**Pull request by OmranAbdelkafi**
## Description
Format and improve help menu
Change encoding progress from number of frame to percentage of progress
## Performance 
No impact
## Issue related
#779
#945

### CODE REVIEW

**Comment by EwoutH** [comment]
Percentage looks nice! I would still like to know the framecount however.

Maybe we can display it as `Encoding   42.00% (420/1000)`?

File: Source/App/EncApp/EbAppProcessCmd.c [hunk 1]
```diff
@@ -1277,7 +1279,24 @@ AppExitConditionType process_output_stream_buffer(EbConfig *config, EbAppContext
 #else
             //++frame_count;
             if (!(header_ptr->flags & EB_BUFFERFLAG_IS_ALT_REF))
+            {
+#if FRAME_OUTPUT
                 fprintf(stderr, "\b\b\b\b\b\b\b\b\b%9d", ++frame_count);
+#else
+                double progress = 0;
+                ++frame_count;
+                progress = ((double)frame_count / config->processed_frame_count) * 100;
```
Comment (hassount) [inline Source/App/EncApp/EbAppProcessCmd.c]: processed_frame_count should be frames_to_be_encoded

**Comment by luigi311** [comment]
Should we start changing names to match up with other encoders, such as enc-mode over to preset

**Comment by silverbacknet** [comment]
> 
> 
> Should we start changing names to match up with other encoders, such as enc-mode over to preset that way we dont have to change around the double dash commands any more and break any existing implementations again when/if we decided to switch over to more standard names in the future after this patch has been merged.

aomenc uses "cpu-used" in the same way as svt's enc-mode, while the x26x family uses "preset", so the need for a cpu-used synonym is probably more pressing. It be fairly simple to map a set of superfast-to-placebo presets to enc-modes, the commandline parsing just isn't set up for that at all yet.

**Comment by kylophone** [comment]
My only concern is that command line parsing is still not being done with `getopt`. Is there a reason for this?

**Comment by luigi311** [comment]
We can make it backwards compatible by using the method I specified on #779

**Comment by kylophone** [comment]
> We can make it backwards compatible by using the method I specified on #779

Yeah. We'll have to. Don't publish the old options in the readme or help/usage message, but if someone uses an old option, accept it but also print a loud depreciation warning.

**Comment by luigi311** [comment]
Looks like theres also not a --version option

**Comment by luigi311** [comment]
Heres a quick example of how we can support all 3 commands so we will not break existing scripts but move over to something new. This supports
```bash
SvtAv1EncApp --help   *new long method*
SvtAv1EncApp -h       *new short method*
SvtAv1EncApp -help    *legacy support*
```
https://github.com/luigi311/SVT-AV1/blob/32af03d4638858795e4de2096094356174aabf5c/Source/App/EncApp/EbAppConfig.c#L1633

**Comment by silverbacknet** [comment]
I think going with getopt_long_only() with long & short options duplicated to some single short option would be the way to go. getopt does take care of some of the overhead and would really simplify that file. SVT already doesn't support getopt-style short-args (either combined args or no space between arg and extra data), so making everything a long arg would work fine.

**Comment by hassount** [comment]
@OmranAbdelkafi Closing this for now, we'll need to open a new one for the new interface to be ported from cs2

**Pull request closed without merging**